openapi-spec/swagger.yaml

```diff
@@ -0,0 +1,534 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+swagger: '2.0'
+info:
+  version: "v1beta"
+  title: OpenSDS Orchestration API
+  description: OpenSDS Orchestration API to manage workflows.
+  termsOfService: 'https://www.linuxfoundation.org/terms'
+  contact:
+    name: OpenSDS Support
+    url: 'https://opensds.slack.com'
+    email: support@opensds.io
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+host: opensds
+schemes:
+  - https
+  - http
+produces:
+  - application/json
+consumes:
+  - application/json
+paths:
+  /:
+    get:
+      tags:
+        - API versions
+      description: Lists information for all Orchestration API versions.
+      responses:
+        '200':
+          description: OK
+          schema:
+            type: object
+            properties:
+              versions:
+                type: array
+                items:
+                  $ref: '#/definitions/VersionSpec'
+          examples:
+            application/json:
+              - name: v1beta
+                status: CURRENT
+                updated: '2017-07-10T14:36:58.014Z'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+  '/{apiVersion}':
+    parameters:
+      - $ref: '#/parameters/apiVersion'
+    get:
+      tags:
+        - API versions
+      description: Gets version details by specified API version.
+      responses:
+        '200':
+          description: OK
+          schema:
+            type: object
+            properties:
+              version:
+                $ref: '#/definitions/VersionSpec'
+          examples:
+            application/json:
+              name: v1beta
+              status: SUPPORTED
+              updated: '2017-04-10T14:36:58.014Z'
+        '404':
+          $ref: '#/responses/HTTPStatus404'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+
+  '/v1beta/{tenantId}/orchestration/services':
+    parameters:
+      - $ref: '#/parameters/tenantId'
+    get:
+      tags:
+        - Service Definitions
+      description: Lists information for all Service Defintions.
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/ServiceDefs'
+        '401':
+          $ref: '#/responses/HTTPStatus401'
+        '403':
+          $ref: '#/responses/HTTPStatus403'
+        '500':
+          $ref: '#/responses/HTTPStatus500'
+    post:
+      tags:
+        - Service Definitions
+      description: Registers a Service Definition.
+      parameters:
+        - name: Definition name
+          in: body
+          schema:
+            $ref: '#/definitions/ServiceDefsReq'
```

Comment:
tenant_id or user_id is not required in request body

Reply:
Done